### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by justinegeffen**
Added license info, reorganized the Handbook intro and added a link to the "how to" page for visibility.

Addresses: https://github.com/mattermost/mattermost-handbook/issues/704

### CODE REVIEW

File: README.md
Comment (jasonblais) [inline README.md]: It's not just for paid staff. There are handbook entries for the contributor community as well, for instance: https://handbook.mattermost.com/contributors/contributors/ways-to-contribute

File: README.md
Comment (jasonblais) [inline README.md:7]: Propose having this line on its own paragraph, so it stands out more easily.

File: README.md
Comment (jasonblais) [inline README.md]: For a future iteration, it looks like we don't really use this mindset as much, and it may not be needed if every page is considered an evolving, work-in-progress document as mentioned in the introduction.

**Review by jasonblais — COMMENTED** [review]
Thanks Justine! Left a couple of small comments.

File: README.md
Comment (justinegeffen) [inline README.md]:
```suggestion
The Mattermost Handbook is a continual work-in-progress or Mattermost staff (people who are paid to work on our open source software), contributor community, and end user community. We use it to document the processes that underpin our company behind our software. Its purpose is to **iteratively increase clarity** for Mattermost staff and community co-building the future of our company and our offerings, and provide a single-source of truth.
```

File: README.md
Comment (justinegeffen) [inline README.md:7]:
```suggestion

Want to contribute to the Handbook? Start here: [How to update the Handbook](company/how-to-guides-for-staff/how-to-update-handbook.md).
```

File: README.md
Comment (justinegeffen) [inline README.md]:
```suggestion
On some pages, you'll see reference to [1% and 50% drafts](company/about-mattermost/mindsets.md#drafts-at-1-50-99). This is used to express how complete our thinking is around the published content, and indicate whether fundamental changes may be made. 
```

**Comment by justinegeffen** [comment]
Thanks, @jasonblais! I've done some minor rewriting as a first pass. LMKWYT.

File: README.md
Comment (jasonblais) [reply README.md]: I would adjust the sentence to read `... for Mattermost staff (people who are paid to work on our open source software), contributor community and end user community.

(since not everyone in our community are contributors, but pretty much everyone are 'users' - whether users of our product, our APIs, or perhaps our handbook :) )

**Comment by jasonblais** [comment]
@justinegeffen I left one proposal, but otherwise the changes look good to me :+1:

**Comment by justinegeffen** [comment]
Thanks, @jasonblais! All changes applied. Thanks for the awesome feedback. :)

**Review by jasonblais — APPROVED** [review]
Thank you!

**Comment by jasonblais** [comment]
@justinegeffen did you want a second review, or is this good to merge?

**Comment by justinegeffen** [comment]
Good to merge, thank you! :D

**Pull request merged**
